syftlib/src/main/java/org/openmined/syft/network/SignallingClient.kt

```diff
@@ -0,0 +1,81 @@
+package org.openmined.syft.network
+
+import io.reactivex.Flowable
+import io.reactivex.processors.PublishProcessor
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.json
+import okhttp3.*
+import java.util.concurrent.TimeUnit
+
+private const val SOCKET_CLOSE_CLIENT = 1000
+private const val SOCKET_CLOSE_ERROR = 1001
+
+internal class SignallingClient(
+    private val workerId: String,
+    private val url: String,
+    private val keepAliveTimeout: Int = 20000
+    ) {
+    private lateinit var request: Request
+    private lateinit var client: OkHttpClient
+    private lateinit var webSocket: WebSocket
+    private val syftSocketListener = SyftSocketListener()
+
+    private val statusPublishProcessor: PublishProcessor<NetworkMessage> = PublishProcessor.create<NetworkMessage>()
+
+    fun start(): Flowable<NetworkMessage> {
+        client = OkHttpClient.Builder()
+            .pingInterval(keepAliveTimeout.toLong(), TimeUnit.MILLISECONDS)
+            .build()
+        request = Request.Builder()
+            .url(url)
+            .build()
+        connect()
+        return statusPublishProcessor.onBackpressureBuffer()
+    }
+
+    /**
+     * Send the data over the Socket connection to PyGrid
+     */
+    fun send(type: String, data: JsonObject) {
+        val message = json {
+            "type" to type
+            "data" to data.content.toMutableMap().replace("workerId", JsonPrimitive(workerId))
+        }.toString()
+
+        webSocket.send(message)
+
+        statusPublishProcessor.offer(NetworkMessage.MessageSent)
```

Comment:
It should publish it when the message call succeeds right? Something like response = webSocket.send()

If response == true => publish ?